Sources/RInAppMessaging/Router.swift

```diff
@@ -137,4 +241,97 @@ internal class Router: RouterType {
+    }
+
+    private func updateFrame(targetView: UIView, tooltipView: TooltipView, superview: UIView, position: TooltipBodyData.Position) {
+        guard targetView.superview != nil else {
+            return
+        }
+        let targetViewFrame = superview.convert(targetView.frame, from: targetView.superview)
+
+        switch position {
+        case .topCenter:
+            tooltipView.frame.origin = CGPoint(x: targetViewFrame.midX - tooltipView.frame.width / 2.0,
+                                               y: targetViewFrame.origin.y - tooltipView.frame.height - UIConstants.Tooltip.targetViewSpacing)
+        case .topLeft:
+            tooltipView.frame.origin = CGPoint(x: targetViewFrame.minX - tooltipView.frame.width,
+                                               y: targetViewFrame.origin.y - tooltipView.frame.height - UIConstants.Tooltip.targetViewSpacing)
+        case .topRight:
+            tooltipView.frame.origin = CGPoint(x: targetViewFrame.maxX,
+                                               y: targetViewFrame.origin.y - tooltipView.frame.height - UIConstants.Tooltip.targetViewSpacing)
+        case .bottomLeft:
+            tooltipView.frame.origin = CGPoint(x: targetViewFrame.minX - tooltipView.frame.width,
+                                               y: targetViewFrame.maxY + UIConstants.Tooltip.targetViewSpacing)
+        case .bottomRight:
+            tooltipView.frame.origin = CGPoint(x: targetViewFrame.maxX,
+                                               y: targetViewFrame.maxY + UIConstants.Tooltip.targetViewSpacing)
+        case .bottomCenter:
+            tooltipView.frame.origin = CGPoint(x: targetViewFrame.midX - tooltipView.frame.width / 2.0,
+                                               y: targetViewFrame.maxY + UIConstants.Tooltip.targetViewSpacing)
+        case .left:
+            tooltipView.frame.origin = CGPoint(x: targetViewFrame.minX - tooltipView.frame.width - UIConstants.Tooltip.targetViewSpacing,
+                                               y: targetViewFrame.midY - tooltipView.frame.height / 2.0)
+        case .right:
+            tooltipView.frame.origin = CGPoint(x: targetViewFrame.maxX + UIConstants.Tooltip.targetViewSpacing,
+                                               y: targetViewFrame.midY - tooltipView.frame.height / 2.0)
+        }
+    }
+}
+
+// MARK: - ViewListenerObserver
+extension Router {
+    func viewDidChangeSuperview(_ view: UIView, identifier: String) {
+        // unused
+    }
+
+    func viewDidMoveToWindow(_ view: UIView, identifier: String) {
+        // unused
+    }
+
+    func viewDidGetRemovedFromSuperview(_ view: UIView, identifier: String) {
+        displayedTooltips[identifier]?.removeFromSuperview()
+        displayedTooltips[identifier] = nil
+    }
+
+    func viewDidUpdateIdentifier(from: String?, to: String?, view: UIView) {
+        if let oldIdentifier = from, displayedTooltips[oldIdentifier] != nil {
+            // update or remove (TBC)
```

Comment:
Is this comment still needed?

Reply:
Yes. But I can remove it and create a ticket instead